gnssanalysis/gn_io/igslog.py

```diff
@@ -101,38 +138,76 @@ def find_recent_logs(logs_glob_path: str, rnx_glob_path=None) -> _pd.DataFrame:
     return recent_logs_df
 
 
+_REGEX_VERSION_1 = _re.compile(rb"""(site log\))""")
+_REGEX_VERSION_2 = _re.compile(rb"""(site log v2)""")
+
+
+def determine_log_version(data: bytes) -> str:
+    """Given the byes object that results from reading an IGS log file, determine the version ("v1.0" or "v2.0")
+
+    :param bytes data: IGS log file bytes object to determine the version of
+    :return str: Return the version number: "v1.0" or "v2.0" (or "Unknown" if file does not conform to standard)
+    """
+    # check for version:
+
+    result_v1 = _REGEX_VERSION_1.search(data)
+    if result_v1:
+        return "v1.0"
+
+    result_v2 = _REGEX_VERSION_2.search(data)
+    if result_v2:
+        return "v2.0"
+
+    raise LogVersionError("Log file does not conform to any known IGS version")
+
+
 def parse_igs_log(filename_array: _np.ndarray) -> _np.ndarray:
     """Parses igs log and outputs ndarray with parsed data
-    Expects ndarray of the form [CODE DATE PATH]"""
-    file_code, __, file_path = filename_array
+
+    :param _np.ndarray filename_array: Metadata on input log file. Expects ndarray of the form [CODE DATE PATH]
+    :return _np.ndarray: Returns array with data from the IGS log file parsed
+    """
+    file_code, _, file_path = filename_array
 
     with open(file_path, "rb") as file:
         data = file.read()
 
+    try:
+        version = determine_log_version(data)
+    except LogVersionError as e:
+        logger.warning(f"Error: {e}, skipping parsing the log file")
```

Comment:
in this case, when the exception is raised, the code will still continue even if the version is uknown, 
I am not sure what do we want to do. 
one option is to stop the function (either re- `raise` 
```python
except LogVersionError as e:
    logger.warning(...)
   raise
```
or just an empty return
```python
except .....
     logger.warning(....)
    return
```

Reply:
I've gone with returning None